App/Components/Avatar.tsx

```diff
@@ -49,96 +51,83 @@ class Avatar extends React.Component<Props, State> {
   }
 
   render () {
-    let width: string | number = 100
-    let height: string | number = width
-    const w = this.props.style && this.props.style.width
-    const h = this.props.style && this.props.style.height
-    if (w) {
-      width = w
-      height = w
-    } else if (h) {
-      width = h
-      height = h
+    const { style, icon, target, local, started, color } = this.props
+
+    const defaultSize = 100
+    let width: string | number = defaultSize
+    let height: string | number = defaultSize
+    if (style) {
+      if (style.width) {
+        width = height = style.width
+      } else if (style.height) {
+        width = height = style.height
+      }
     }
+    const { borderRadius: radius } = this.state
 
-    const { peerId, localPeerId, nodeStarted, localAvatar } = this.props
-
-    const isLocalUser = (!peerId && localPeerId) || peerId === localPeerId
-
-    const peerIdToUse = peerId || localPeerId
-    const uri = peerIdToUse ? `${Config.RN_TEXTILE_CAFE_GATEWAY_URL}/ipns/${peerIdToUse}/avatar/large` : undefined
-
-    const widthNumber = typeof width === 'number' ? width as number : undefined
-    const heightNumber = typeof height === 'number' ? height as number : undefined
+    if (icon || !started || !target) {
+      return (
+        <Icon
+          style={{ ...(this.props.style || {}), width, height, borderRadius: radius }}
+          name={icon || 'question-circle'}
+          size={height as number}
+          color={s.COLOR_GREY_LIGHT}
+          onLayout={this.onImageLayout}
+        />
+      )
+    }
 
-    if (isLocalUser && nodeStarted && localAvatar && widthNumber) {
-      // Render TextileImage
+    // local means the target belongs to the local node
+    if (local) {
       return (
-        <View style={{ width, height, borderRadius: this.state.borderRadius, overflow: 'hidden' }}>
+        <View style={{ ...(this.props.style || {}), width, height, borderRadius: radius, overflow: 'hidden' }}>
           <TextileImage
-            style={{ ...(this.props.style || {}), width, height, borderRadius: this.state.borderRadius }}
-            target={localAvatar}
+            style={{ ...(this.props.style || {}), width, height, borderRadius: radius }}
+            target={target}
             index={0}
-            forMinWidth={widthNumber}
+            forMinWidth={width as number}
```

Comment:
Going to have some problems here if width was set as a string, like `'50%'`, for example. That's why it used to use `widthNumber` and wouldn't render the `TextileImage` unless width is a number.

Reply:
Ok, cool. I'll add a check back in